### PR TITLE
Consolidate e2e test container

### PR DESCRIPTION
**Pull request by gctucker**
Reuse the Docker image from `docker/api` in end-to-end tests rather than keeping a copy in `docker/test` with minor tweaks.

Also use the same `requirements-dev.txt` file for both the `api` and `test` containers to rely on the Docker build cache and avoid doing 2 builds.  I haven't timed it precisely but the GitHub action duration seems to have reduced from 3min to 2min.

### CODE REVIEW

File: docker/api/requirements-dev.txt [hunk 1]
```diff
@@ -1,3 +1,8 @@
 -r requirements-tests.txt
 pycodestyle==2.8.0
 pylint==2.12.2
+pytest==6.2.5
+pytest-asyncio==0.16.0
+pytest-dependency==0.5.1
+pytest-order==1.0.1
+httpx==0.23.3
```
Comment (JenySadadia) [inline docker/api/requirements-dev.txt:8]: These should be added to `requirements-tests.txt` as these packages are specific for testing.

File: test-docker-compose.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ services:
     build:
       context: 'docker/api'
       args:
-        - REQUIREMENTS=${REQUIREMENTS:-requirements.txt}
+        - REQUIREMENTS=${REQUIREMENTS:-requirements-dev.txt}
```
Comment (JenySadadia) [inline test-docker-compose.yaml:14]: What is the need of changing the requirements file here?

Comment (gctucker) [reply test-docker-compose.yaml:14]: It's explained in the commit message.

File: docker/api/requirements-dev.txt (hunk 1 above)
Comment (gctucker) [reply docker/api/requirements-dev.txt:8]: OK

**Review by JenySadadia — APPROVED** [review]
LGTM

**Pull request merged**